### PR TITLE
Capture trace of error reporting thread and identify with boolean flag

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Captures the trace of the crashed thread, which was previously omitted, and to identify it with a boolean flag in the serialised JSON.

## Changeset

- Add the current thread trace during serialisation
- Conditionally write `errorReportingThread` boolean flag by checking `crashed` flag

Note: the thread name is not captured due to known limitations with how KSCrash captures this information.

## Tests

Added unit tests for general thread serialisation, and for checking that the current thread is now captured. Tested with dashboard.

## Review

For the submitter, initial self-review:

- [x] Commented on code changes inline explain the reasoning behind the approach
- [x] Reviewed the test cases added for completeness and possible points for discussion
- [x] A changelog entry was added for the goal of this pull request
- [x] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [x] Initial review of the intended approach, not yet feature complete
  - [x] Structural review of the classes, functions, and properties modified
  - [x] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [x] Consistency across platforms for structures or concepts added or modified
- [x] Consistency between the changeset and the goal stated above
- [x] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [x] Usage friction - is the proposed change in usage cumbersome or complicated?
- [x] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [x] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [x] Thoroughness of added tests and any missing edge cases
- [x] Idiomatic use of the language

### CODE REVIEW

File: Source/BugsnagCrashReport.m [hunk 1]
```diff
@@ -579,29 +580,38 @@ - (NSArray *)serializeThreadsWithException:(NSMutableDictionary *)exception {
...
+        [self serialiseThread:bugsnagThreads thread:thread backtrace:backtrace crashedThread:isCrashedThread];
+    }
+    return bugsnagThreads;
+}
 
-            for (NSDictionary *frame in backtrace) {
+- (void)serialiseThread:(NSMutableArray *)bugsnagThreads
+                 thread:(NSDictionary *)thread
+              backtrace:(NSArray *)backtrace
+          crashedThread:(BOOL)isCrashedThread {
```
Comment (snmaynard) [inline Source/BugsnagCrashReport.m]: I would be careful to call this the reportingThread or the currentThread to emphasize that its not the crashing thread in all cases.

**Comment by fractalwrench** [comment]
Labelling as WIP as the mazerunner scenario appears to have failed on Travis CI, and the build queue is currently blocked which means I'm unlikely to investigate this further by the end of today :(
